#0	BEHAVIOR_NONE,        //!< Robot currently not in any behavior; software freeze, controls disabled.
#1	BEHAVIOR_FREEZE,      //!< All joints frozen in current position.
#2	BEHAVIOR_STAND_PREP,  //!< Robot assumes a balanced posture, preparing for Stand.
#3	BEHAVIOR_STAND,       //!< Statically stable stand.
#4	BEHAVIOR_WALK,        //!< Dynamically stable walk.
#5	BEHAVIOR_STEP,        //!< Statically stable, single-step walk.
#6	BEHAVIOR_MANIPULATE,  //!< Statically stable stand, with upper body joints controlled by user.
#7	BEHAVIOR_USER,        //!< Setpoints and gains set by user.
#8	BEHAVIOR_CALIBRATE,   //!< Robot moves through various calibration motions, returns to Freeze.
#9	BEHAVIOR_SOFT_STOP,   //!< "Safe" falling behavior, robot tucks its arms and legs and uses soft position gains
#
# bdi_mode  : maps to BEHAVIOR_ enum from Atlas interface, for -1 keep current behavior
# upper_body_control: value of ub_cntrl_ flag in interface : 0=BDI controls, 1=We control, -1=don't change (i.e. preserve the prior setting)
# switch_mode: when to change active controllers 0=disable, 1=on command, 2=on feedback, 3=force reset of all controllers, 4=force reset but maintain current
# active_controllers: list of named controllers that we will start and stop when switching behavior control modes
#
atlas_controller:
    allowed_control_modes:
        - "none"
        - "stop"
        - "freeze"
        - "stand_prep"
        - "stand"
        - "walk"
        - "step"
        - "manipulate"
        - "whole_body"
        - "dance"
        - "calibrate"
        - "soft_stop"
        - "stand_manipulate"
        - "walk_manipulate"
        - "step_manipulate"
        - "manipulate_gravity"
        - "manipulate_inverse_dynamics"
        - "manipulate_limits"
        - "manipulate_limits_stabilized"
        - "manipulate_friction_with_gravity"
        - "manipulate_compliant_impedance"
        - "manipulate_stiff_impedance"
        - "manipulate_observer_impedance"
        - "dance_impedance"
    control_mode_to_controllers:
#  "none" should be the first controller defined (No controllers active)
        none:
            bdi_mode: 1
            upper_body_control:  0
            switch_mode: 3
            active_joint_controllers: ["whole_body_tracking_controller", "neck_tracking_controller"]
            active_robot_controllers: []
            nominal_hydraulic_pressure: 500
        stop:
#   Force a reset of all controllers and footsteps (trajectory controllers reset to hold current position)
            bdi_mode: -1
            upper_body_control:  -1
            switch_mode: 3
            active_joint_controllers:  ["left_arm_traj_controller", "right_arm_traj_controller", "torso_traj_controller", "neck_traj_controller", "lower_body_tracking_controller"]
            active_robot_controllers: []
            nominal_hydraulic_pressure: 1500
        freeze:
# Lock all joints and do a hard reset
            bdi_mode: 1
            upper_body_control:  0
            switch_mode: 3
            active_joint_controllers: [ "whole_body_tracking_controller", "neck_tracking_controller"]
            active_robot_controllers: []
            nominal_hydraulic_pressure: 500
        stand_prep:
            bdi_mode: 2
            upper_body_control:  0
            switch_mode: 2
            active_joint_controllers: [ "neck_traj_controller", "whole_body_tracking_controller"]
            active_robot_controllers: []
            nominal_hydraulic_pressure: 1500
        stand:
            bdi_mode: 3
            upper_body_control:  0
            switch_mode: 2
            active_joint_controllers: [ "whole_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: [ "footstep_controller" ]
            nominal_hydraulic_pressure: 1250
        stand_manipulate:
            bdi_mode: 3
            upper_body_control:  1
            switch_mode: 2
            active_joint_controllers: [ "left_arm_traj_controller", "right_arm_traj_controller", "torso_traj_controller", "lower_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: [ "footstep_controller" ]
            nominal_hydraulic_pressure: 2600
        walk:
            bdi_mode: 4
            upper_body_control:  0
            switch_mode: 1
            active_joint_controllers: [ "whole_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: [ "footstep_controller" ]
            nominal_hydraulic_pressure: 1600
        walk_manipulate:
            bdi_mode: 4
            upper_body_control:  1
            switch_mode: 1
            active_joint_controllers: [ "left_arm_traj_controller", "right_arm_traj_controller", "torso_traj_controller", "lower_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: [ "footstep_controller" ]
            nominal_hydraulic_pressure: 2300
        step:
            bdi_mode: 5
            upper_body_control:  0
            switch_mode: 1
            active_joint_controllers: [ "whole_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: [ "footstep_controller" ]
            nominal_hydraulic_pressure: 2000
        step_manipulate:
            bdi_mode: 5
            upper_body_control:  1
            switch_mode: 1
            active_joint_controllers: [ "left_arm_traj_controller", "right_arm_traj_controller", "torso_traj_controller", "lower_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: [ "footstep_controller" ]
            nominal_hydraulic_pressure: 2300
        manipulate:
            bdi_mode: 6
            upper_body_control:  -1
            switch_mode: 1
            active_joint_controllers: ["left_arm_traj_controller", "right_arm_traj_controller", "torso_traj_controller", "lower_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: ["pelvis_traj_controller" ]
            nominal_hydraulic_pressure: 2600
        whole_body:
            bdi_mode: 7
            upper_body_control:  -1
            switch_mode: 1
            active_joint_controllers: ["whole_body_traj_controller", "neck_traj_controller"]
            active_robot_controllers: []
            nominal_hydraulic_pressure: 2600
        dance:
            bdi_mode: 7
            upper_body_control:  -1
            switch_mode: 1
            active_joint_controllers: ["left_arm_traj_controller", "right_arm_traj_controller", "torso_traj_controller", "left_leg_traj_controller", "right_leg_traj_controller", "neck_traj_controller"]
            active_robot_controllers: []
            nominal_hydraulic_pressure: 2600
        calibrate:
            bdi_mode: 8
            upper_body_control:  0
            switch_mode: 2
            active_joint_controllers: [ "whole_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: []
            nominal_hydraulic_pressure: 2500
        soft_stop:
            bdi_mode: 9
            upper_body_control:  0
            switch_mode: 2
            active_joint_controllers: ["whole_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: []
            nominal_hydraulic_pressure: 500
        manipulate_gravity:
            bdi_mode: 6
            upper_body_control:  -1
            switch_mode: 1
            active_joint_controllers: ["left_arm_traj_controller", "right_arm_traj_controller", "torso_traj_controller", "lower_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: ["pelvis_traj_controller", "gravity_compensation_controller" ]
            nominal_hydraulic_pressure: 2600
        manipulate_inverse_dynamics:
            bdi_mode: 6
            upper_body_control:  -1
            switch_mode: 1
            active_joint_controllers: ["left_arm_traj_controller", "right_arm_traj_controller", "torso_traj_controller", "lower_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: ["pelvis_traj_controller", "inverse_dynamics_controller" ]
            nominal_hydraulic_pressure: 2600
        manipulate_limits:
            bdi_mode: 6
            upper_body_control:  -1
            switch_mode: 1
            active_joint_controllers: ["left_arm_joint_limit_traj_controller", "right_arm_joint_limit_traj_controller", "torso_traj_controller", "lower_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: [ "pelvis_traj_controller" ]
            nominal_hydraulic_pressure: 2600
        manipulate_limits_stabilized:
            bdi_mode: 7
            upper_body_control:  -1
            switch_mode: 1
            active_joint_controllers: ["left_arm_joint_limit_traj_controller", "right_arm_joint_limit_traj_controller", "lower_body_relax_controller", "neck_traj_controller"]
            active_robot_controllers: [ ]
            nominal_hydraulic_pressure: 2600
        manipulate_friction_with_gravity:
            bdi_mode: 6
            upper_body_control:  -1
            switch_mode: 1
            active_joint_controllers: ["left_arm_traj_controller", "right_arm_traj_controller", "torso_traj_controller", "lower_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: ["pelvis_traj_controller", "gravity_compensation_controller", "friction_compensation_controller" ]
            nominal_hydraulic_pressure: 2600
        manipulate_compliant_impedance:
            bdi_mode: 6
            upper_body_control:  -1
            switch_mode: 1
            active_joint_controllers: ["left_arm_traj_impedance_controller", "right_arm_traj_impedance_controller", "torso_traj_controller", "lower_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: ["pelvis_traj_controller", "left_arm_compliant_joint_impedance_controller", "right_arm_compliant_joint_impedance_controller" ]
            nominal_hydraulic_pressure: 2600
        manipulate_stiff_impedance:
            bdi_mode: 6
            upper_body_control:  -1
            switch_mode: 1
            active_joint_controllers: ["left_arm_traj_hybrid_controller", "right_arm_traj_hybrid_controller", "torso_traj_controller", "lower_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: ["pelvis_traj_controller", "left_arm_stiff_joint_impedance_controller", "right_arm_stiff_joint_impedance_controller" ]
            nominal_hydraulic_pressure: 2600
        manipulate_observer_impedance:
            bdi_mode: 6
            upper_body_control:  -1
            switch_mode: 1
            active_joint_controllers: ["left_arm_traj_hybrid_controller", "right_arm_traj_hybrid_controller", "torso_traj_controller", "lower_body_tracking_controller", "neck_traj_controller"]
            active_robot_controllers: ["pelvis_traj_controller", "left_arm_observer_joint_impedance_controller", "right_arm_observer_joint_impedance_controller" ]
            nominal_hydraulic_pressure: 2600
        dance_impedance:
            bdi_mode: 7
            upper_body_control:  -1
            switch_mode: 1
            active_joint_controllers: ["left_arm_traj_impedance_controller", "right_arm_traj_impedance_controller", "left_leg_traj_impedance_controller", "right_leg_traj_impedance_controller", "torso_traj_controller", "neck_traj_controller"]
            active_robot_controllers: ["left_arm_joint_impedance_controller", "right_arm_joint_impedance_controller", "left_leg_joint_impedance_controller", "right_leg_joint_impedance_controller" ]
            nominal_hydraulic_pressure: 2600

